src/Haskell/Ide/Engine/Transport/JsonStdio.hs

```diff
@@ -34,8 +36,8 @@ jsonStdioTransport cin = do
           putStr $ show (HieError (A.String $ T.pack $ show err))
           loop (cid + 1) stream'
         Just (Right r) -> do
-          writeChan cin (wireToChannel cout cid r)
-          rsp <- readChan cout
+          atomically $ writeTChan cin (wireToChannel cout cid r)
```

Comment:
Even if the plugin is async we block completely at this point, which seems like the wrong thing to do, since we don't gain anything from having async that way. But we should probably try to get the reworked `jsonStdioTransport` before changing that.